### PR TITLE
Add proteomics rules to data production workflow

### PR DESCRIPTION
**Pull request by pmb59**
The goal of this merge request is to include rules for proteomics studies processing (baseline and differential) in Atlas.

The following experiments types will be now supported:
- [x] proteomics_baseline
- [x] proteomics_baseline_dia
- [x] proteomics_differential

Notes:
- `recalculations` operation has not been implemented, only `reprocess`. In principle, further reprocessing analyses should not harm the data, as it implements checks, necessary renames and redecoration with gene name from the latest Ensembl release.
- `get_exptype_contrasts_assays.pl` is failing for experiment type 'proteomics_baseline_dia'. Because we do not need assays/contrasts for baseline proteomics, this failure can be avoided. But ideally the config yaml file in [perl-atlas-modules](https://github.com/ebi-gene-expression-group/perl-atlas-modules) should include this experiment type as one of valid ones for Atlas.
- Although in the old recipe we found `gxa_generate_tracks_for_experiment.sh $expTargetDir`, I have disabled the request to get these outputs as I didn't see any of the track files present  in $ATLAS_PROD for 'proteomics_differential' experiments.

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -348,9 +352,19 @@ def get_checkpoints_cp_atlas_exps(wildcards):
     else:
         return None
 
+def input_round_log2_fold_changes(wildcards):
+    """
+    Ensure rename files has been run before rounding log2 fold changes, for differential proteomics
+    """
+    inputs_files = [ f"{wildcards['accession']}-analytics.tsv.undecorated" ]
+    if experiment_type =='proteomics_differential':
```
Comment (pcm32) [inline Snakefile]:
```suggestion
    if experiment_type == 'proteomics_differential':
```

File: Snakefile [hunk 2]
```diff
@@ -1208,9 +1229,10 @@ rule differential_statistics_rnaseq:
     resources: mem_mb=get_mem_mb
     input:
         config_xml="{accession}-configuration.xml",
-        raw_counts_undecorated="{accession}-raw-counts.tsv.undecorated"
+        raw_counts_undecorated=lambda wildcards: f"{wildcards.accession}-raw-counts.tsv.undecorated" if experiment_type!='proteomics_differential' else 'none_necessary'
```
Comment (pcm32) [inline Snakefile]:
```suggestion
        raw_counts_undecorated=lambda wildcards: f"{wildcards.accession}-raw-counts.tsv.undecorated" if experiment_type != 'proteomics_differential' else 'none_necessary'
```

File: bin/condense_sdrf.pl [hunk 3]
```diff
@@ -310,7 +310,7 @@ sub copy_idf_from_ae {
     my ( $args, $idfFile ) = @_;
     $logger->info( "Copying IDF from $idfFile ..." );
     my $outputDir = $args->{ "output_directory" };
-    `cp $idfFile $outputDir`;
+    `cp -f $idfFile $outputDir`;
```
Comment (pcm32) [inline bin/condense_sdrf.pl:313]: What is the point of this?

File: envs/proteomics_processing.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+channels:
+ - conda-forge
+ - bioconda
+dependencies:
+ - r
+ - perl-atlas-modules >=0.3.1
```
Comment (pcm32) [inline envs/proteomics_processing.yaml:6]: aren't we using the container in the end for this? maybe remove this file or amend if not needed.

**Review by pcm32 — APPROVED** [review]
LGTM in general. I trust you on the general logic and that this is working well somewhere. Some minor suggestions.

File: envs/proteomics_processing.yaml (hunk 4 above)
Comment (pmb59) [reply envs/proteomics_processing.yaml:6]: Currently we use containers only for ensembl-update-env:amm1.1.2

File: bin/condense_sdrf.pl (hunk 3 above)
Comment (pmb59) [reply bin/condense_sdrf.pl:313]: This is to mitigate the following error in rule `get_magetab_for_experiment`
```
cp: cannot create regular file '[ACC]/[ACC].idf.txt': Permission denied
```
for some diff proteomics accessions.

**Comment by pmb59** [comment]
Thanks for the review @pcm32  I'll merge soon if I see no errors.

**Comment by pmb59** [comment]
@pcm32 I am merging now. I've added linting via GitHub actions. I'll address linting issues in a different PR

**Pull request merged**